controllers/kubeadmconfig_controller.go

```diff
@@ -330,27 +355,46 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(cluster *capiv1alpha2.Clust
 	return nil
 }
 
-func (r *KubeadmConfigReconciler) getCertificates() (cloudinit.Certificates, error) {
-	//TODO(fp) check what is the expected flow for certificates
-	certificates, _ := NewCertificates()
-
-	return cloudinit.Certificates{
-		CACert:           string(certificates.ClusterCA.Cert),
-		CAKey:            string(certificates.ClusterCA.Key),
-		EtcdCACert:       string(certificates.EtcdCA.Cert),
-		EtcdCAKey:        string(certificates.EtcdCA.Key),
-		FrontProxyCACert: string(certificates.FrontProxyCA.Cert),
-		FrontProxyCAKey:  string(certificates.FrontProxyCA.Key),
-		SaCert:           string(certificates.ServiceAccount.Cert),
-		SaKey:            string(certificates.ServiceAccount.Key),
-	}, nil
+func (r *KubeadmConfigReconciler) getClusterCertificates(ctx context.Context, clusterName, namespace string) (*certs.Certificates, error) {
+	secret := &corev1.Secret{}
+
+	err := r.Get(ctx, types.NamespacedName{Name: fmt.Sprintf("%s-certs", clusterName), Namespace: namespace}, secret)
+	if err != nil {
+		return nil, err
+	}
+
+	return certs.MapToCertificates(secret.Data), nil
 }
 
-// SetupWithManager TODO
-func (r *KubeadmConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&cabpkv1alpha2.KubeadmConfig{}).
-		Complete(r)
+func (r *KubeadmConfigReconciler) createClusterCertificates(ctx context.Context, clusterName string, config *cabpkv1alpha2.KubeadmConfig) (*certs.Certificates, error) {
+	certificates, err := certs.NewCertificates()
+	if err != nil {
+		return nil, err
+	}
+
+	secret := &corev1.Secret{
```

Comment:
I think this will need to be 1 secret per cert to support cert-manager creating the certs: See `https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/74#issuecomment-518357317

Reply:
Integration with cert-manager would likely need additional work and design, I'm not sure I would block this change on that.

Reply:
If I run cert-manager (or anything else) and use it to pre-create the certs, and only then create the cluster it should work fine as long as the certificates are 1 per secret and named correctly. 

Integrating cert-manager with CAPI or this provider would require additional design, but that shouldn't stop people using it un-integrated

Reply:
Do we think that this is something that should block this PR? Or should we open an issue to make the secret format compatible with cert-manger?

Reply:
+1 to creating an issue and handling any potential changes in post.

Reply:
This was agreed to as part of the issue, please see the comment history on the issue.

Reply:
https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/135 - happy to re word if its unclear or I missed something.